crates/bcr-wdc-treasury-service/src/debit/service.rs

```diff
@@ -117,8 +168,94 @@ where
     }
 }
 
+async fn monitor_quote(
+    qid: String,
+    ebill_id: String,
+    wlt: impl Wallet,
+    repo: impl Repository,
+    wdc: impl WildcatService,
+    interval: tokio::time::Duration,
+) {
+    loop {
+        tokio::time::sleep(interval).await;
+        let result = wlt.mint(qid.clone()).await;
+        let Ok(status) = result else {
+            tracing::warn!("Failed to mint quote {qid}: {result:?}");
+            continue;
+        };
+        if !matches!(status, cashu::MintQuoteState::Paid) {
+            tracing::info!("Quote {qid} is not paid yet, retrying...");
+            continue;
+        }
+        let result = wdc.deactivate_keyset_for_ebill(&ebill_id).await;
+        if let Err(e) = result {
+            tracing::warn!(
+                "Failed to deactivate keyset for ebill {ebill_id} after minting quote {qid}: {e}"
+            );
+            continue;
+        }
+        let result = repo.delete_quote(qid.clone()).await;
+        match result {
+            Ok(_) => {
+                tracing::info!("Successfully deactivated keyset for ebill {ebill_id} after minting quote {qid}");
```

Comment:
should this be `successfully deleted`? Or is this basically the second part of deactivation (deactivate + delete)